### PR TITLE
Support for GNGNS Strings

### PR DESCRIPTION
**Pull request by bmurray**
I also rolled in a change to allow SNR values to be Floats. I have a receiver that always reports SNR with a float value.

### CODE REVIEW

File: gngns.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	ManualGNGNS            = "M"
+	SimulatorGNGNS         = "S"
+)
+
+type GNGNS struct {
+	BaseSentence
+	Time       Time
+	Latitude   float64
+	Longitude  float64
+	Mode       []string
```
Comment (icholy) [inline gngns.go]: What if we did `Mode` like this instead:

``` go
type ModeGNGNS string

func (m ModeGNGNS) String() string {
  return string(m)
}

func (m ModeGNGNS) IsNoFix() bool {
  return strings.Contains(string(m), "N")
}

func (m ModeGNGNS) IsAutonomous() bool {
  return strings.Contains(string(m), "A")
}

// etc ...
```

File: parser.go [hunk 2]
```diff
@@ -62,6 +62,27 @@ func (p *parser) EnumString(i int, context string, options ...string) string {
 	p.SetErr(context, s)
 	return ""
 }
+func (p *parser) EnumStrings(i int, context string, options ...string) []string {
```
Comment (icholy) [inline parser.go]: A better name would be `EnumChars`.

File: parser.go [hunk 3]
```diff
@@ -62,6 +62,27 @@ func (p *parser) EnumString(i int, context string, options ...string) string {
...
+	}
+	strs := []string{}
+	for i := 0; i < len(s); i++ {
+		r := string(s[i])
+		for _, o := range options {
+			if o == r {
+				strs = append(strs, o)
+			}
+		}
+	}
```
Comment (icholy) [inline parser.go]:
``` go
for _, r := range s {
    rs := string(r)
    for _, o := range options {
        // etc ...
    }
}
```

File: gngns.go (hunk 1 above)
Comment (bmurray) [reply gngns.go]: That would certainly be simpler. However, the values are positional based on which constellation has the appropriate fix. Eg. My receiver gives me "AAN", which tells me GPS and Glonass are up, but Galileo is down. Perhaps a convenience method for a simple test would be in order, in addition to positional values.

Comment (bmurray) [reply gngns.go]: How about something like this? 

```go
func (g GNGNS) IsMode(modes ...string) bool {
        for _, m := range g.Mode {
                for _, md := range modes {
                        if m == md {
                                return true
                        }
                }
        }
        return false
}
```

Comment (icholy) [reply gngns.go]: @bmurray ah that makes more sense. In that case, let's leave it the way it is.

**Comment by icholy** [comment]
Considering that the mode values are contextual, I don't see the use-case for the `IsMode` method.

**Comment by icholy** [comment]
@bmurray I'm ready to merge it if you revert back to the original RP (and the tests are passing).

**Comment by icholy** [comment]
@adrianmo the change to a `float64` will be a breaking change. We'll have to bump the major version :/

**Comment by bmurray** [comment]
Thinking more about it, the *GSV values should be Int64, as that's what's in the spec. The library shouldn't compromise for broken chips. 

I've also removed the IsMode from the GNGNS string. Test's are all passing on my end.

**Comment by bmurray** [comment]
Hm, they're commented now, and it runs against current Go. I don't know what those other errors are related to; it seems like Travis-CI is having issues pulling golint ?

**Comment by icholy** [comment]
@bmurray yeah, this is related to https://github.com/adrianmo/go-nmea/pull/41

**Comment by icholy** [comment]
Thanks for the contribution!

**Pull request merged**